executor/src/state.rs

```diff
@@ -59,3 +62,25 @@ impl ExecutionIndices {
         transaction_index == self.next_transaction_index
     }
 }
+
+impl Ord for ExecutionIndices {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        if self.next_certificate_index == other.next_certificate_index {
+            if self.next_batch_index == other.next_batch_index {
+                self.next_transaction_index
+                    .cmp(&other.next_transaction_index)
+            } else {
+                self.next_batch_index.cmp(&other.next_batch_index)
+            }
+        } else {
+            self.next_certificate_index
+                .cmp(&other.next_certificate_index)
+        }
+    }
```

Comment:
You could do that ... or you could write
```rust
        (
            self.next_certificate_index,
            self.next_batch_index,
            self.next_transaction_index,
        )
            .cmp(&(
                other.next_certificate_index,
                other.next_batch_index,
                other.next_transaction_index,
            ))
```
which will [generate. the same thing](https://doc.rust-lang.org/src/core/tuple.rs.html#77-83)

Reply:
ha that is much better